code/modules/hydroponics/grown/melon.dm

```diff
@@ -39,6 +40,36 @@
 /obj/item/food/grown/watermelon/make_dryable()
 	return //No drying
 
+/obj/item/food/grown/watermelon/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/kitchen/spoon))
+		return ..()
+
+	var/melon_pulp_count = 1
+	if(seed)
+		melon_pulp_count += round(seed.potency / 25)
+
+	user.balloon_alert(user, "scooped out [melon_pulp_count] pulp")
+	for(var/i in melon_pulp_count)
+		new /obj/item/food/watermelonmush(user.loc)
+
+	var/melon_armour
+	var/melon_chestplate_chance = (max(0, seed.potency - 50) / 50)
+	if (!prob(melon_chestplate_chance))
+		if(seed.resistance_flags & FIRE_PROOF)
+			melon_armour = new /obj/item/clothing/head/helmet/durability/watermelon/fire_resist
+		else
+			melon_armour = new /obj/item/clothing/head/helmet/durability/watermelon
+		to_chat(user, span_notice("You hollow the melon into a chestplate with [I]."))
+	else
+		if(seed.resistance_flags & FIRE_PROOF)
+			melon_armour = new /obj/item/clothing/suit/armor/durability/watermelon/fire_resist
+		else
+			melon_armour = new /obj/item/clothing/suit/armor/durability/watermelon
```

Comment:
please invert this check. so like instead of doing `if(!prob(chance))` do `if(prob(chance))` and put the suit stuff first. makes things easier to read/mentally process